### PR TITLE
MockMaker API cleanup

### PR DESCRIPTION
**Pull request by AndreasTu**
Moved MockMakerId to spock.mock package.
Added release notes for mock maker feature.

Try to ease flaky InvokingMocksFromMultipleThreads spec

PollingConditions change: Fix failing test when tests are executed with non english locale.
The formatting of the elapsedTime would not match the pattern in the test.

### CODE REVIEW

File: spock-specs/src/test/groovy/org/spockframework/smoke/mock/InvokingMocksFromMultipleThreads.groovy [hunk 1]
```diff
@@ -24,9 +24,11 @@ import org.spockframework.mock.TooManyInvocationsError
 import spock.lang.*
 
 class InvokingMocksFromMultipleThreads extends Specification {
-  def numThreads = 10
+  private static final int WAIT_TIME_S = 200
+
+  def numThreads = WAIT_TIME_S
```
Comment (leonard84) [inline spock-specs/src/test/groovy/org/spockframework/smoke/mock/InvokingMocksFromMultipleThreads.groovy:29]: Creating 200 threads could take some time on slower machines.

At least for Java 21 we could use virtual threads.

```groovy
  @AutoCleanup("shutdownNow")
  ExecutorService executorService = createExecutorService()

  ExecutorService createExecutorService() {
    Jvm.current.java21Compatible ? Executors."newVirtualThreadPerTaskExecutor"() : Executors.newCachedThreadPool {new Thread(it).tap { it.daemon = true } }
  }
```

File: spock-specs/src/test/groovy/org/spockframework/smoke/mock/JavaStubs.groovy [hunk 2]
```diff
@@ -141,6 +141,24 @@ class JavaStubs extends Specification {
     person.phoneNumber == "6789"
   }
 
+  def "can stub final methods with mockito with closure"() {
+    FinalMethodPerson person = Stub(mockMaker: MockMakers.mockito) {
```
Comment (leonard84) [inline spock-specs/src/test/groovy/org/spockframework/smoke/mock/JavaStubs.groovy:145]: Unfortunetely, this specs is not consitent in its use, but new features should include proper block labels.

```suggestion
    given:
    FinalMethodPerson person = Stub(mockMaker: MockMakers.mockito) {
```

File: spock-specs/src/test/groovy/org/spockframework/smoke/mock/JavaStubs.groovy [hunk 3]
```diff
@@ -166,6 +184,14 @@ class JavaStubs extends Specification {
     e.message.contains("global")
   }
 
+  def "no static type specified"() {
+    when:
+    Stub()
+    then:
```
Comment (leonard84) [inline spock-specs/src/test/groovy/org/spockframework/smoke/mock/JavaStubs.groovy:190]:
```suggestion

    then:
```

**Review by leonard84 — CHANGES_REQUESTED** [review]

File: spock-core/src/main/java/spock/mock/IMockMakerSettings.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+@Beta
+public interface IMockMakerSettings {
+
+  /**
+   * Creates a simple {@code IMockMakerSettings} object with only an ID.
+   *
+   * @param id the mock maker ID.
+   * @return the settings object
+   */
+  static IMockMakerSettings simple(MockMakerId id) {
```
[the remark this answers is not in the document]
Comment (AndreasTu) [reply spock-core/src/main/java/spock/mock/IMockMakerSettings.java]: Ich have changed the name to `settingsFor()`.

**Review by leonard84 — APPROVED** [review]

**Pull request merged**